site/app/controllers/student/SubmissionController.php

```diff
@@ -269,8 +270,14 @@ private function changeRequestStatus() {
         }
     }
 
-    private function showHomeworkPage() {
-        $gradeable_id = (isset($_REQUEST['gradeable_id'])) ? $_REQUEST['gradeable_id'] : null;
+    /**
+     * @Route("/{_semester}/{_course}/student/{gradeable_id}")
```

Comment:
should this also include `@Route('/{_semester}/{_course}/student/{gradeable_id}/{gradeable_version}` with gradeable_version being optional (and would default to latest if not there).

why is gradeable_id optional here?

Reply:
Oh. At first I thought it would be backward compatible - Classic Router doesn't pass args to functions and we need a default value. Now it seems useless.